### PR TITLE
Added D401 support to kubernetes provider

### PR DESCRIPTION
**Pull request by guptarohit**
related: https://github.com/apache/airflow/issues/37256

Fixes D401 lint issues in kubernetes provider.

- [x] airflow/providers/cncf/kubernetes/callbacks.py
- [x] airflow/providers/cncf/kubernetes/operators/custom_object_launcher.py
- [x] airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py
- [x] airflow/providers/cncf/kubernetes/resource_convert/configmap.py
- [x] airflow/providers/cncf/kubernetes/resource_convert/env_variable.py
- [x] airflow/providers/cncf/kubernetes/resource_convert/secret.py

### CODE REVIEW

**Review by dirrao — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/callbacks.py
Comment (hussein-awala) [inline airflow/providers/cncf/kubernetes/callbacks.py:1]: Not sure if we can apply D401 on callbacks (in general and not just in this module)

**Review by hussein-awala — COMMENTED** [review]
Let's wait for other reviews

File: airflow/providers/cncf/kubernetes/callbacks.py [hunk 1]
```diff
@@ -41,15 +41,15 @@ class KubernetesPodOperatorCallback:
...
+        """Call this after creating the sync client.
 
         :param client: the created `kubernetes.client.CoreV1Api` client.
         """
         pass
 
     @staticmethod
     def on_pod_creation(*, pod: k8s.V1Pod, client: client_type, mode: str, **kwargs) -> None:
-        """Callback method called after creating the pod.
+        """Call this after creating the pod.
```
Comment (Taragolis) [inline airflow/providers/cncf/kubernetes/callbacks.py]: I guess these might be more close to the original docstring title:
- `Invoke the callback after the pod has been created.`
- `Invoke the callback following the creation of the pod.`

Comment (guptarohit) [reply airflow/providers/cncf/kubernetes/callbacks.py]: I see, good suggestion. Thanks.
what about using `Invoke this callback after the pod has been created.`

File: airflow/providers/cncf/kubernetes/callbacks.py
Comment (guptarohit) [reply airflow/providers/cncf/kubernetes/callbacks.py:1]: okay! so, shall I skip the callback file?
I'm not very sure why we shouldn't apply D401 on callbacks, could you please possible the reason for it?

File: airflow/providers/cncf/kubernetes/callbacks.py (hunk 1 above)
Comment (hussein-awala) [reply airflow/providers/cncf/kubernetes/callbacks.py]: sounds good

Comment (guptarohit) [reply airflow/providers/cncf/kubernetes/callbacks.py]: done, pushed new changes

**Review by hussein-awala — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/callbacks.py
Comment (potiuk) [reply airflow/providers/cncf/kubernetes/callbacks.py:1]: I am good with it as well? what problem do you see @hussein-awala :) ?

Comment (potiuk) [reply airflow/providers/cncf/kubernetes/callbacks.py:1]: I see you approved, so resolving.

**Pull request merged**